sdk/textanalytics/Azure.AI.TextAnalytics/tests/TextAnalyticsClientLiveTests.cs

```diff
@@ -397,8 +397,7 @@ public async Task ExtractKeyPhrasesTest()
             TextAnalyticsClient client = GetClient();
             string document = "My cat might need to see a veterinarian.";
 
-            Response<IReadOnlyCollection<string>> response = await client.ExtractKeyPhrasesAsync(document);
-            IReadOnlyCollection<string> keyPhrases = response.Value;
+            KeyPhraseCollection keyPhrases = await client.ExtractKeyPhrasesAsync(document);
```

Comment:
(Sorry if I'm just not understanding c#) I'm a little confused why the way you're accessing keyPhrases hasn't changed in the rest of the test. Since you're adding warnings and the extracted key phrases to KeyPhraseCollection, wouldn't you have to call .keyPhrases on your KeyPhraseCollection to get the extracted key phrases, or do you only have to do that with warnings?

Reply:
So `KeyPhraseCollection` inherits from `ReadOnlyCollection<string>`, so it contains the list of phrases already in the type, which allows me to iterate over it.
You are right that for warnings, I do need to do `.Warnings`.

By doing it this way, users don't have to do an extra call to get to the data they really care about, which makes it easier to consume. This is also the reason why the name of the class ends with `Collection`. In .net land, it tells users that the type has that inheritance.

Does that make sense?

Reply:
yep, thanks for explaining!